controllers/openshift_controller.go

```diff
@@ -888,6 +888,13 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// Revert the feature gate status configmap to default
+	err = r.FeatureGates.RevertFeatureGateStatusConfigMapToDefault(context.TODO())
```

Comment:
I believe it would be cleaner, and simpler too, to just remove the cm here.  As I said elsewhere, I'm far from convinced that we should insist on this configmap's existence in the first place, but it definitely serves no purpose when there's even no KataConfig instance, right?

Reply:
Yes, I'm also not sure why do we need to update the user's CM? This might look strange from the user's PoV having some settings changed. We should not mix default values and user's values. Whoever create it shouldn't be the one responsible for deleting?

Not a k8s expert, and I was not following all the discussions here closely, so I'm probably missing something.

Reply:
That is a very good point, we should avoid muddying the notion of resource ownership to avoid problems like we had a while ago with `KataConfig.spec.kataMonitorImage` which was basically an OSC-owned field within a user-owner `KataConfig` instance.

However, my takeaway from the latest round of discussions is that we're now leaning towards the controller _not_ requiring the existence of any of the feature-gating configmaps.  This solves the ownership issue quite neatly by putting the configmap resources wholly under user control.